### DISCUSSION TITLE
Validation fails using vet but it succeeds using export

### DISCUSSION THREAD

**Discussion by ahmed711** [cue-lang/cue#1954] [Q&A]
Hello All, 
I am trying to validate the sum of a list but it fails, I guess the validation run before the statement is executed and that is why it fails 

``` 
import "list"

{
    students: [
     {
        name: "Emanuel",
        grade: 20
     },
     {
        name: "Amin",
        grade: 30
     }
    ]
    students: [...#student]
    #student: { 
        name : string
        grade: int
    }
    _GradeSumValidation: true & list.Sum([ for student in students { student.grade }]) == 50
}
``` 

I created this [example ](https://cuelang.org/play/?id=zq3_NADxIrU#cue@export@cue) to demonstrate the problem 

if I run this with ` cue vet ` I receive an error but if I run it with ` cue export ` it passes 

is there any way that would delay the validation till after the statement is executed if there is the reason why it fails

Thanks

**Comment by slewiskelly** [comment]
What is the error you observed? 

When attempting to reproduce, it correctly succeeds using both `export` and `vet`:

```console
$ cat students.cue
import "list"

{
    students: [
        {
            name:  "Emanuel"
            grade: 20
        },
        {
            name:  "Amin"
            grade: 30
        },
    ]
    students: [...#student]
    #student: {
        name:  string
        grade: int
    }
    _GradeSumValidation: true & list.Sum([ for student in students {student.grade}]) == 50
}

$ cue vet students.cue
$ echo $status
0

$ cue export students.cue
{
    "students": [
        {
            "name": "Emanuel",
            "grade": 20
        },
        {
            "name": "Amin",
            "grade": 30
        }
    ]
}
```

**Reply by slewiskelly** [reply]
Version information might also be pertinent.

```console
$ cue version
cue version v0.4.3 darwin/arm64
```

**Reply by ahmed711** [reply]
Thanks for your reply @slewiskelly 

I have my schema in a different file 
so I have 2 files 

`students.json` 
```
{
    "students": [
        {
            "name":  "Emanuel",
            "grade": 20
        },
        {
            "name":  "Amin",
            "grade": 30
        }
    ]
}
```

and `students.cue` 
```
import "list"

students: [...#student]
#student: {
    name:  string
    grade: int
    }
_GradeSumValidation: true & list.Sum([ for student in students {student.grade}]) == 50
```

the error I receive is 
```
_GradeSumValidation: conflicting values false and true:
    ./students.cue:8:22
    ./students.cue:8:29 
```

my cue version is `cue version v0.4.3 linux/amd64`

**Reply by slewiskelly** [reply]
Appears similar to https://github.com/cue-lang/cue/issues/1926.

**Reply by ahmed711** [reply]
if I modify the GradeSumValidation to 
` _GradeSumValidaton: 50 & list.Sum([ for student in students {student.grade}]) `

I receive conflicting value 
``` 
_GradeSumValidation: conflicting values 50 and 0 
```
So I think the validation takes place before json is loaded into schema

**Reply by ahmed711** [reply]
As a work around I used the following 

```
import "list"

students: [...#student]
#student: {
    name:  string
    grade: int & >0
    }
_totalGrade: list.Sum([ for student in students {student.grade}])
_gradeValidation: ( 50 | 0 ) & _totalGrade
```
